python/ray/serve/controller.py

```diff
@@ -580,9 +583,25 @@ def run_graph(import_path: str, deployment_override_options: List[Dict]):
     app = build(graph)
 
     # Override options for each deployment
-    for options_dict in deployment_override_options:
-        name = options_dict["name"]
-        app.deployments[name].set_options(**options_dict)
+    for options in deployment_override_options:
+        name = options["name"]
+
+        # Merge graph-level and deployment-level runtime_envs
+        ray_actor_options = options.get("ray_actor_options")
+        if ray_actor_options is None:
+            if app.deployments[name].ray_actor_options:
+                ray_actor_options = app.deployments[name].ray_actor_options
+            else:
+                ray_actor_options = {}
```

Comment:
```
ray_actor_options = options.get("ray_actor_options") or app.deployments[name].ray_actor_options or {}
```

Reply:
I like the concision, but I think this may introduce a bug. Suppose the user explicitly specifies an empty dictionary for their `ray_actor_options` in the config, but they specify non-empty `ray_actor_options` in the code. Then, this statement will evaluate to the code's non-empty options instead of the config's empty options since an empty `dict` is falsy.

I've added a check for this in `test_put_new_rest_api` in `test_serve_head.py`.